app/controllers/lab_tests_controller.rb

```diff
@@ -80,20 +88,55 @@ def destroy
 
   private
 
-  # Use callbacks to share common setup or constraints between actions.
+  def handle_success_response
+    flash[:notice] = t('.success')
+    redirect_to(@health_record)
+  end
+
+  def handle_error_response(message = nil)
+    flash.now[:alert] = message if message
+    load_error_dependencies
+    render :new, status: :unprocessable_entity
+  end
+
+  def determine_redirect_path
+    @health_record || @lab_test
+  end
+
+  def load_error_dependencies
+    return if @lab_test.biomarker_id.blank?
+
+    @selected_biomarker = Biomarker.find(@lab_test.biomarker_id)
+    @reference_ranges = @selected_biomarker.reference_ranges
+  end
+
+  def set_biomarkers
+    @biomarkers = Biomarker.all
+  end
+
   def set_lab_test
     @lab_test = LabTest.find(params[:id])
   end
 
-  def build_lab_test
-    @lab_test = current_user.lab_tests.build(lab_test_params)
+  def save_records
+    @health_record.save && @lab_test.save
   end
 
-  # Only allow a list of trusted parameters through.
   def lab_test_params
-    params
-      .require(:lab_test)
-      .permit(:user_id, :biomarker_id, :value, :unit, :reference_range_id, :recordable_type, :recordable_id, :notes,
-              :created_at, :updated_at)
+    params.require(:lab_test).permit(
+      :biomarker_id,
+      :value,
+      :unit,
+      :reference_range_id,
+      :notes
+    )
+  end
+
+  def determine_user
```

Comment:
```suggestion
  def get_user
```

Simpler and much more common naming for this kind of logic.

Reply:
the method determine_user is better because it shows there is logic and decisions involved, and it fits Ruby's style of using clear names. The method get_user is less preferred because "get" is not common in Ruby, it doesn't show there is logic, and it sounds like a simple accessor. But, sure I can rename it, if you want.

Reply:
Agreed with getter in Rails.

Than it should be `set_user`, if you want make it "Rails way".

Reply:
done

Reply:
no, you are wrong .... I need to back for my var naming. when I tried to pass rubocop I got this:

`Inspecting 98 files
.........C........................................................................................

Offenses:

app/controllers/lab_tests_controller.rb:138:7: C: Naming/AccessorMethodName: Do not prefix reader method names with get_.
  def get_user
      ^^^^^^^^

98 files inspected, 1 offense detected

Tip: Based on detected gems, the following RuboCop extension libraries might be helpful:
  * rubocop-capybara (https://rubygems.org/gems/rubocop-capybara)

You can opt out of this message by adding the following to your config (see https://docs.rubocop.org/rubocop/extensions.html#extension-suggestions for more options):
  AllCops:
    SuggestExtensions: false`

Reply:
with a determine_user all passed

Reply:
@VoldemLive didn't get what are you talking about.

I said about SET_user and you tried GET_user and got an error.
Rubocop throws an error about getters, but my point is about another name `set_user`, and NOT `get_user`.

Reply:
you are right, with set_user all test are passed